### ISSUE TITLE
apm-agent-php destroys response code in the curl

### ISSUE THREAD

**Issue by real420og** [elastic/apm-agent-php#629]
by default in case of timeout curl return response code 28 - curl_errno
but apm-agent-php predefine curl response code in case of timeout and set 0 - curl_errno

here always set result code 0 https://github.com/elastic/apm-agent-php/blob/main/src/ext/tracer_PHP_part.c#L170
but curl can return other response codes, such as 28 for CURLE_OPERATION_TIMEDOUT.

php-script example:

```
$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, 'http://timeout.timeout/10/'); // returns a response after 10 seconds
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_TIMEOUT, 5); // waiting for a response 5 seconds

$response = curl_exec($ch);
$errorCode = curl_errno($ch);

curl_close($ch);
```

when APM-AGENT disabled errorCode contain 28
but when APM-AGENT enabled errorCode contain 0

this behavior breaks packages that are linked to the curl response code

---
relevant for version v1.4.2

**Comment by waynegemmell** [comment]
+1. We really need this to work asap.

**Comment by SergeyKleyman** [comment]
Fixed in #707
The fix was released in [v1.5.2](https://github.com/elastic/apm-agent-php/releases/tag/v1.5.2)
